keel-web/src/main/kotlin/com/netflix/spinnaker/config/KeelConfigurationFinalizer.kt

```diff
@@ -33,8 +35,7 @@ class KeelConfigurationFinalizer(
 
   @PostConstruct
   fun registerResourceSpecSubtypes() {
-    resourceHandlers
-      .map { it.supportedKind }
+    (resourceHandlers.map { it.supportedKind } + specMigrators.map { it.input })
```

Comment:
This allows us to still accept submissions of obsolete spec types so users are not forced to update their delivery configs right away. Long term we can consider the possibility of indicating that things could be updated somehow. I can see that being either in the response to the upsert endpoint, or the UI (or both).